common/docker/client.go

```diff
@@ -0,0 +1,81 @@
+package docker
+
+import (
+	"github.com/fsouza/go-dockerclient"
+	. "github.com/weaveworks/weave/common"
+)
+
+// An observer for container events
+type ContainerObserver interface {
+	ContainerDied(ident string) error
+}
+
+type Client struct {
+	client *docker.Client
+}
+
+// NewClient creates a new Docker client
+func NewClient(apiPath string) (*Client, error) {
+	dc, err := docker.NewClient(apiPath)
+	if err != nil {
+		return nil, err
+	}
+
+	c := Client{
+		client: dc,
+	}
+	return &c, nil
+}
+
+// Start starts the client
+func (c *Client) Start(apiPath string) error {
+	client, err := docker.NewClient(apiPath)
+	if err != nil {
+		Error.Printf("[docker] Unable to connect to Docker API on %s: %s", apiPath, err)
+		return err
+	}
+
+	env, err := client.Version()
+	if err != nil {
+		Error.Printf("[docker] Unable to connect to Docker API on %s: %s", apiPath, err)
+		return err
+	}
+	Info.Printf("[docker] Using Docker API on %s: %v", apiPath, env)
+	return nil
+}
+
+// AddObserver adds an observer for docker events
+func (c *Client) AddObserver(ob ContainerObserver) error {
+	events := make(chan *docker.APIEvents)
+	if err := c.client.AddEventListener(events); err != nil {
+		Error.Printf("[docker] Unable to add listener to Docker API: %s", err)
+		return err
+	}
+
+	go func() {
+		for event := range events {
+			switch event.Status {
+			case "die":
+				id := event.ID
+				ob.ContainerDied(id)
+			}
+		}
+	}()
+	return nil
+}
+
+// IsContainerRunning returns true if the container is up and running
+func (c *Client) IsContainerRunning(idStr string) bool {
+	running := true
+
+	// check with Docker whether the container is really running
+	if container, err := c.client.InspectContainer(idStr); err != nil {
+		if _, notThere := err.(*docker.NoSuchContainer); notThere {
+			running = false
+		}
+	} else {
+		running = container.State.Running
+	}
+
+	return running
+}
```

Comment:
I don't understand the logic here; why do we return `true` when inspecting the container fails with something other than `NoSuchContainer`?